packages/app/src/components/Root/ApplicationProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import { useContext } from 'react';
+
+import { ErrorBoundary } from '@backstage/core-components';
+
+import DynamicRootContext from '../DynamicRoot/DynamicRootContext';
+
+export const ApplicationProvider = ({
+  children,
+}: React.PropsWithChildren<{}>) => {
+  const { mountPoints } = useContext(DynamicRootContext);
+  const providers = mountPoints['application/provider'] ?? [];
+  if (providers.length === 0) {
+    return children;
+  }
+  return providers.map(({ Component }, index) => {
+    return (
+      <ErrorBoundary
+        // eslint-disable-next-line react/no-array-index-key
+        key={index}
+      >
+        <Component>{children}</Component>
+      </ErrorBoundary>
+    );
+  });
```

Comment:
This solution works only if you have one provider.

Multiple providers would render the children multiple times, which isn't right.